core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SecurityScanGithubWorkflowProbeTest.java

```diff
@@ -0,0 +1,196 @@
+
+        Files.write(workflowFile, List.of(
+            "name: Test Security Scan Job",
+            "jobs:",
+            "  security-scan:",
+            "    uses: not-the-workflow-definition-we-are-looking-for@v1"
+        ));
+        when(ctx.getScmRepository()).thenReturn(repo);
+
+        final ProbeResult result = probe.apply(plugin, ctx);
+        assertThat(result.status()).isEqualTo(ResultStatus.FAILURE);
+        assertThat(result.message()).isEqualTo("GitHub workflow security scan is not configured in the plugin");
+    }
+
+    @Test
+    void shouldSucceedIfSecurityScanIsConfigured() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Path workflowPath = Files.createDirectories(repo.resolve(".github/workflows"));
+        final Path workflowFile = Files.createFile(workflowPath.resolve("jenkins-security-scan.yaml"));
+
+        Files.write(workflowFile, List.of(
+            "name: Test Security Scan Job",
+            "jobs:",
+            "  this-is-a-valid-security-scan:",
+            "    uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml"
+        ));
+        when(ctx.getScmRepository()).thenReturn(repo);
+
+        final ProbeResult result = probe.apply(plugin, ctx);
+        assertThat(result.status()).isEqualTo(ResultStatus.SUCCESS);
+        assertThat(result.message()).isEqualTo("GitHub workflow security scan is configured in the plugin");
+    }
+
+    @Test
+    void shouldSucceedIfWorkflowIsConfigured() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Path workflowPath = Files.createDirectories(repo.resolve(".github/workflows"));
+        final Path workflowFile = Files.createFile(workflowPath.resolve("jenkins-security-scan.yaml"));
+
+        Files.write(workflowFile, List.of(
+            "name: Test Security Scan Job",
+            "jobs:",
+            "  security-scan:",
+            "    uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml"
```

Comment:
Just to make sure that the version is not a problem for the code.

```suggestion
            "    uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml@v42"
```